### PR TITLE
Signup

### PR DESCRIPTION
**Pull request by TiboldiRoberto**
What does it fix?

Add:

- SignUpView
- SignUpFormView
- CustomCheckbox
- ValidationAccountView

### CODE REVIEW

File: DeUrgenta/Components/General/CustomCheckboxView.swift [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+    var body: some View {
+        HStack {
+            VStack {
+                Button(action: {
+                    if isChecked {
+                        isChecked = false
+                    } else {
+                        isChecked = true
+                    }
```
Comment (MihaiPantiru) [inline DeUrgenta/Components/General/CustomCheckboxView.swift]: Is this equivalent to isChecked = !isChecked ?

File: DeUrgenta/Features/Authentication/SignUp/View/SignUpView.swift [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    func signUpViewDidTapSignUp(_ view: SignUpView)
+}
+
+struct SignUpView: View {
+    var delegate: SignUpViewDelegate
+
+    var body: some View {
+        ScrollView {
+            VStack(alignment: .leading) {
+                Text(AppStrings.Authentication.signUpHeaderTItle.localized())
```
Comment (MihaiPantiru) [inline DeUrgenta/Features/Authentication/SignUp/View/SignUpView.swift]: Typo: **signUpHeaderTItle**

File: DeUrgenta/Features/Authentication/SignUp/View/ValidationAccountView.swift [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+                .lineSpacing(5)
+                .padding(.horizontal)
+
+            Spacer()
+        }
+        .padding()
+    }
+}
+
+struct ValidationAccountView_Previews: PreviewProvider {
```
Comment (MihaiPantiru) [inline DeUrgenta/Features/Authentication/SignUp/View/ValidationAccountView.swift]: Do we need the preview code?

File: DeUrgenta/Features/Authentication/SignUp/ViewModel/SignUpViewModel.swift [hunk 4]
```diff
@@ -0,0 +1,68 @@
+import Foundation
+
+final class SignUpViewModel: ObservableObject {
+    @Published var firstname: String = ""
```
Comment (MihaiPantiru) [inline DeUrgenta/Features/Authentication/SignUp/ViewModel/SignUpViewModel.swift]: firstName
lastName

**Review by MihaiPantiru — CHANGES_REQUESTED** [review]
Great work @TiboldiRoberto. I left you some comments regarding minor changes/clarifications.

File: DeUrgenta/Components/General/CustomCheckboxView.swift (hunk 1 above)
Comment (TiboldiRoberto) [reply DeUrgenta/Components/General/CustomCheckboxView.swift]: Yes, it's equivalent.

**Review by MihaiPantiru — APPROVED** [review]

**Pull request merged**